composeml/label_search.py

```diff
@@ -0,0 +1,93 @@
+from pandas import isnull
+
+from composeml.utils import format_number, is_finite_number
+
+
+class ExampleSearch:
+    """A label search based on the number of examples.
+
+    Args:
+        expected_count (int): The expected number of examples to find.
+    """
+
+    def __init__(self, expected_count):
+        self.expected_count = format_number(expected_count)
+        self.reset_count()
+
+    @property
+    def is_complete(self):
+        """Whether the search has found the expected number of examples."""
+        return self.actual_count >= self.expected_count
+
+    @property
+    def is_finite(self):
+        """Whether the expected number of examples is a finite number."""
+        return is_finite_number(self.expected_count)
+
+    def is_valid_labels(self, labels):
+        """Whether the label values are not null."""
+        return not any(map(isnull, labels.values()))
+
+    def reset_count(self):
+        """Reset the internal count of actual labels."""
+        self.actual_count = 0
+
+    def update_count(self, labels):
```

Comment:
does this method need the `labels` parameter?

Reply:
Search classes do require this parameter as an [interface](https://github.com/FeatureLabs/compose/pull/130/files#diff-22bf41f653d21f0ca3dd15dadf5e01e9R305) in the search process.

```python
search.update_count(labels)
```

The intention is to make an interface requirement, but not a requirement for inherited methods to use the parameter. The example search class does not need this parameter, whereas the label search class does.